client/packages/system/src/Name/ListView/ImportProperties/UploadTab.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { FC, useState } from 'react';
+import Papa, { ParseResult } from 'papaparse';
+import { InlineProgress, Typography, Upload } from '@common/components';
+import { ImportPanel } from './ImportPanel';
+import { useTranslation } from '@common/intl';
+import {
+  Grid,
+  Stack,
+  Link,
+  FileUtils,
+  PropertyNode,
+  useNotification,
+  NamePropertyNode,
+} from '@openmsupply-client/common';
+
+import { ImportRow } from './PropertiesImportModal';
+import { importFacilitiesPropertiesToCsv } from '../utils';
+import { processProperties } from 'packages/system/src/utils';
+import { FacilityNameRowFragment } from '../../api/operations.generated';
+
+interface UploadTabProps {
+  setFacilityProperties: React.Dispatch<React.SetStateAction<ImportRow[]>>;
+  setErrorMessage: (value: React.SetStateAction<string>) => void;
+  facilities: FacilityNameRowFragment[] | undefined;
+  onUploadComplete: () => void;
+  properties: NamePropertyNode[] | undefined;
+}
+interface ParsedImport {
+  id: string;
+  [key: string]: string | undefined;
+}
+
+enum FacilitiesColumn {
+  CODE = 0,
```

Comment:
Maybe more fields helpful here? I'm not sure

Reply:
I think these are the main ones you'd want